src/core.js

```diff
@@ -101,52 +124,105 @@ async function pruneNonCriticalCssLauncher ({
+            if (response.url === url) {
+              debuglog('RESPONSE URL: ' + response.url) // TODO: interceptedRequest.url is a function in puppeteer > 0.12
+              if (pageLoadSkipTimeout) {
+                debuglog('pageLoadSkipTimeout injected on dom creation')
+                // This is crucial for the evaluate to work. If this is not set we got an error:
+                // ERROR Error: Protocol error (Runtime.callFunctionOn): Cannot find context with specified id undefined
+                // Maybe this should be evaluated on low power machines if they need 100ms
+                await page.waitFor(500)
+                page
+                  .evaluate(pageLoadSkipTimeoutFunc, {
+                    pageLoadSkipTimeout
+                  })
+                  .then(message => {
+                    if (!_hasLoaded) {
+                      // when pageLoadSkipTimeout is reached after dom request
+                      // don't resolve when page already loaded completely
+                      debuglog(
+                        'pageLoadSkipTimeout - page load waiting ABORTED after ' +
+                          pageLoadSkipTimeout / 1000 +
+                          's. '
+                      )
+                      return resolve('pageLoadSkipTimeout')
+                    }
+                  })
+                  .catch(err => {
+                    if (!err.message.includes('Target closed')) {
+                      debuglog('page.evaluate - ERROR', err)
+                      _hasError = err
+                      return reject(err)
+                    }
+                  })
+              }
+            }
+          })
         }
       })
 
       debuglog('page load start')
       // set a higher number than the timeout option, in order to make
       // puppeteer’s timeout _never_ happen
-      let waitingForPageLoad = true
-      const loadPagePromise = page.goto(url, { timeout: timeout + 1000 })
-      if (pageLoadSkipTimeout) {
-        await Promise.race([
-          loadPagePromise,
-          new Promise(resolve => {
-            // instead we manually _abort_ page load after X time,
-            // in order to deal with spammy pages that keep sending non-critical requests
-            // (tracking etc), which would otherwise never load.
-            // With JS disabled it just shouldn't take that many seconds to load what's needed
-            // for critical viewport.
-            setTimeout(() => {
-              if (waitingForPageLoad) {
-                debuglog(
-                  'page load waiting ABORTED after ' +
-                    pageLoadSkipTimeout / 1000 +
-                    's. '
-                )
-                resolve()
-              }
-            }, pageLoadSkipTimeout)
+      const loadPageResponse = new Promise((resolve, reject) => {
+        page
+          .goto(url, {
+            timeout: timeout + 1000,
+            waitUntil: 'networkidle'
```

Comment:
Hmm, I tried this value before but it leads some urls to get stuck, if they continuously send requests for tracking and similar. I'm not sure it is a better default for Penthouse. What was your reasoning behind this change?

Reply:
This is supposed to your other questions. 
I did the window.stop() in the executed page tab because I want the load event get fired. In this case the timeout wins the race with a correct "window load event". I did some testing with that but after all the changes I made I wouldnt bet it wouldn't run without it.

Reply:
I actually see that Penthouse uses `networkidle2`, but specified in the puppeteer.launch call [here](https://github.com/pocketjoso/penthouse/blob/master/src/index.js#L51)... which according to the existing documentation seems wrong - although they might have moved it in a more recent version.
So all in all, I would be in favour of testing the `networkidle2` value here - but unless it is related to your changes I suggest for you _not_ to do it (in this PR), and I can isolate and test just that change myself (later).